### PR TITLE
Replace non-ascii characters in headers to be in line with the service.

### PR DESCRIPTION
**Pull request by AsafMah**
### Fixed
* Replace non-ascii characters in headers to be in line with the service.

### CODE REVIEW

File: ingest/src/test/java/com/microsoft/azure/kusto/ingest/E2ETest.java [hunk 1]
```diff
@@ -122,6 +122,7 @@ public static void setUp() throws IOException {
 
         ConnectionStringBuilder engineCsb = ConnectionStringBuilder.createWithAadApplicationCredentials(System.getenv("ENGINE_CONNECTION_STRING"), appId,
                 appKey, tenantId);
+        engineCsb.setUserNameForTracing("Java_E2ETest_ø");
```
Comment (yihezkel) [inline ingest/src/test/java/com/microsoft/azure/kusto/ingest/E2ETest.java:125]: This is great for testing that sending a non-ASCII char doesn't cause ingestion to **fail**. But can we use a getter to ensure in this test that it properly translates the char to a "?"?

File: data/src/main/java/com/microsoft/azure/kusto/data/ClientImpl.java [hunk 2]
```diff
@@ -307,6 +307,13 @@ private Map<String, String> generateIngestAndCommandHeaders(ClientRequestPropert
                 activityId, activityId, activityId, clientRequestId);
         headers.put("x-ms-activitycontext", activityContext);
 
+        // replace non-ascii characters in header values with '?'
+        for (Map.Entry<String, String> entry : headers.entrySet()) {
```
Comment (yihezkel) [inline data/src/main/java/com/microsoft/azure/kusto/data/ClientImpl.java:311]: nit: I like how you did it as well, but I think this is a great scenario for using the stream API.

**Review by yihezkel — APPROVED** [review]
Approving with suggestions (though 1 is a nit, and I didn't verify the other is even possible)

File: data/src/main/java/com/microsoft/azure/kusto/data/ClientImpl.java [hunk 3]
```diff
@@ -307,6 +307,13 @@ private Map<String, String> generateIngestAndCommandHeaders(ClientRequestPropert
...
+            String value = entry.getValue();
+            if (value != null) {
+                entry.setValue(value.replaceAll("[^\\x00-\\x7F]", "?"));
```
Comment (yihezkel) [inline data/src/main/java/com/microsoft/azure/kusto/data/ClientImpl.java:314]: Suggest using `[^\\x20-\\x7E]`, per https://www.asciitable.com/

File: ingest/src/test/java/com/microsoft/azure/kusto/ingest/E2ETest.java (hunk 1 above)
Comment (AsafMah) [reply ingest/src/test/java/com/microsoft/azure/kusto/ingest/E2ETest.java:125]: Discussed offline

File: data/src/main/java/com/microsoft/azure/kusto/data/ClientImpl.java (hunk 2 above)
Comment (AsafMah) [reply data/src/main/java/com/microsoft/azure/kusto/data/ClientImpl.java]: Done

File: data/src/main/java/com/microsoft/azure/kusto/data/ClientImpl.java (hunk 3 above)
Comment (AsafMah) [reply data/src/main/java/com/microsoft/azure/kusto/data/ClientImpl.java]: The point was to align this to the behavior of the c# SDK.
They don't touch other character, so we won't either.

**Pull request merged**